evm_loader/program/src/entrypoint.rs

```diff
@@ -332,7 +341,14 @@ fn do_call<'a>(
     info!(match exit_reason {
         ExitReason::Succeed(_) => {
             let (applies, logs) = executor.deconstruct();
-            backend.apply(applies, logs, false)?;
+            let logs2 = backend.apply(applies, logs, false)?;
+            for log in logs2 {
+                let ix = on_event(myself_info.key, log)?;
```

Comment:
It's better to use `program_id` here.

Reply:
@s-medvedev But more consistent way is to use myself_info. And, in future, it allows add another contract which receives OnReturn, OnEvent.

Reply:
Ok. Let's leave it that way.